### PR TITLE
Don't expose internal cflags to pkg-config

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

This fixes issue #2951.

## Tasklist
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

**Comment by TheMarex** [comment]
I don't understand why the 32bit build does not work anymore with this, see https://travis-ci.org/Project-OSRM/osrm-backend/jobs/176806721#L1488-L1516

On the one side the cmake for the `osrm-example` seems to detect 32bit but then prints:

> /usr/bin/ld: skipping incompatible /usr/local/lib/libosrm.a when searching for -losrm

**Comment by oxidase** [comment]
@TheMarex the problem was in https://github.com/Project-OSRM/osrm-backend/pull/3330/commits/9abe341c21361b7081c0e174c3aa8b2ea9b33ebd#diff-60ec4631040a17b93814f9da4955c281R36 environment flags were ignored and  x86_64 was used to build example. i also moved flags to .travis to make them visible.

**Comment by TheMarex** [comment]
Had to do some fixes for mason induced weirdness as well. All good now.

@daniel-j-h please review. :bowing_man:

File: .travis.yml [hunk 1]
```diff
@@ -55,17 +54,18 @@ matrix:
...
     # Release Builds
     - os: linux
-      compiler: "mason-release"
+      compiler: "mason-linux-release"
       addons:
         apt:
           sources: ['ubuntu-toolchain-r-test']
           packages: ['libstdc++-5-dev']
-      env: BUILD_TYPE='Release' ENABLE_MASON=ON
+      env: CLANG_VERSION='3.8.1' BUILD_TYPE='Release' ENABLE_MASON=ON
```
Comment (daniel-j-h) [inline .travis.yml]: Why 3.8.1 - 3.9 is last stable

File: .travis.yml [hunk 2]
```diff
@@ -77,7 +77,9 @@ matrix:
 
     - os: linux
       compiler: "gcc-6-release-i686"
-      env: TARGET_ARCH='i686' CCOMPILER='gcc-6' CXXCOMPILER='g++-6' BUILD_TYPE='Release'
+      env: >
```
Comment (daniel-j-h) [inline .travis.yml]: Before reading through the 80 pdf Yaml spec (again), I guess this just allows multi-line env vars?

File: .travis.yml [hunk 3]
```diff
@@ -116,15 +118,23 @@ matrix:
...
-  - if [[ $(uname -s) == 'Darwin' ]]; then sudo mdutil -i off /; fi;
+  - |
+    if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+      export JOBS=$((`nproc` + 1))
+    fi
+  - |
+    if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+      export JOBS=$((`sysctl -n hw.ncpu` + 1))
+      sudo mdutil -i off /
+    fi
```
Comment (daniel-j-h) [inline .travis.yml]: Do you know how the number of procs works on the Travis machines? I think I remember reading somewhere it was reporting the hosts number of procs and should be hardcoded.

File: CMakeLists.txt [hunk 4]
```diff
@@ -240,6 +234,7 @@ if (ENABLE_COVERAGE)
 endif()
 if (ENABLE_SANITIZER)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
+  set(OSRM_CFLAGS "${OSRM_CFLAGS} -fsanitize=address")
```
Comment (daniel-j-h) [inline CMakeLists.txt]: Why do we expose the sanitizer flags to users? Is this needed to build the example?

Comment (TheMarex) [inline CMakeLists.txt]: Yes this is needed if you link a sanitizer enabled libosrm in any library.

File: .travis.yml (hunk 1 above)
Comment (TheMarex) [inline .travis.yml]: Just keeps the current version we use on master. Upgrading should be a separate PR. This fixes a problem where we didn't install clang because this wasn't set.

File: .travis.yml (hunk 3 above)
Comment (TheMarex) [inline .travis.yml]: https://travis-ci.org/Project-OSRM/osrm-backend/jobs/177262393#L1117

Seems to work. I think this was only a problem on container machines not GCE.
